### PR TITLE
Document 'execution_summary' response part of detection APIs

### PR DESCRIPTION
**Pull request by xcrzx**
**Resolves: https://github.com/elastic/security-docs/issues/1759**

Previews:
- [Bulk action API](https://security-docs_1900.docs-preview.app.elstc.co/guide/en/security/master/bulk-actions-rules-api.html) 
- [Find rules API](https://security-docs_1900.docs-preview.app.elstc.co/guide/en/security/master/rules-api-find.html)
- [Get rule API](https://security-docs_1900.docs-preview.app.elstc.co/guide/en/security/master/rules-api-get.html)
- [Update rule API ](https://security-docs_1900.docs-preview.app.elstc.co/guide/en/security/master/rules-api-update.html)

## Summary

Document the `execution_summary` part of the rule schema and mark it as a feature under development

Screenshot 2022-04-28 at 18 03 42
.

### CODE REVIEW

File: docs/detections/api/rules/rules-api-bulk-actions.asciidoc [hunk 1]
```diff
@@ -405,6 +413,8 @@ For `enable`, `disable`, `delete`, `edit`, and `duplicate` actions, a JSON objec
 }
 --------------------------------------------------
 
+<1> dev:[] `execution_summary` is in development and its schema may change in the future.
```
Comment (nastasha-solomon) [inline docs/detections/api/rules/rules-api-bulk-actions.asciidoc:416]:
```suggestion
<1> dev:[] `execution_summary` is under development and its schema may change.
```

**Review by nastasha-solomon — COMMENTED** [review]
Thanks for making this update @xcrzx ! I left one suggestion that can apply across all the entries if you decide to apply it. :)

File: docs/detections/api/rules/rules-api-bulk-actions.asciidoc [hunk 2]
```diff
@@ -386,7 +381,20 @@ For `enable`, `disable`, `delete`, `edit`, and `duplicate` actions, a JSON objec
...
                "actions":[
 
-               ]
+               ],
+               "execution_summary": {                 <1>
+                 "last_execution": {
+                   "date": "2022-03-23T16:06:12.787Z",
+                   "status": "partial failure",
+                   "status_order": 20,
+                   "message": "This rule is attempting to query data from Elasticsearch indices listed in the \"Index pattern\" section of the rule definition, however no index matching",
```
Comment (joepeeples) [inline docs/detections/api/rules/rules-api-bulk-actions.asciidoc:390]: I assume this is just a sample message, but if it's possible to edit the actual message that the API returns, that would be great, too. Also make the same edits to the other APIs in the PR.
```suggestion
                   "message": "This rule is attempting to query data from Elasticsearch indices listed in the \"Index pattern\" section of the rule definition, but no matching index was found.",
```

**Review by joepeeples — COMMENTED** [review]
Suggestion for revising the sample message, otherwise LGTM!

File: docs/detections/api/rules/rules-api-bulk-actions.asciidoc (hunk 2 above)
Comment (benironside) [inline docs/detections/api/rules/rules-api-bulk-actions.asciidoc:390]:
```suggestion
                   "message": "This rule attempted to query data from Elasticsearch indices listed in the \"Index pattern\" section of the rule definition, however no index matching",
```

**Review by benironside — COMMENTED** [review]
👍 Left one minor suggestion

File: docs/detections/api/rules/rules-api-bulk-actions.asciidoc (hunk 1 above)
Comment (xcrzx) [reply docs/detections/api/rules/rules-api-bulk-actions.asciidoc]: Applied your suggestion, thanks!

File: docs/detections/api/rules/rules-api-bulk-actions.asciidoc (hunk 2 above)
Comment (xcrzx) [reply docs/detections/api/rules/rules-api-bulk-actions.asciidoc]: Applied your suggestion, thanks! Will also edit the actual message.

Comment (xcrzx) [reply docs/detections/api/rules/rules-api-bulk-actions.asciidoc]: Applied your suggestion, thanks!

**Pull request merged**